ez/ez_test.go

```diff
@@ -0,0 +1,46 @@
+package ez
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type Config struct {
+	// Path will contain the path to the config file and will be set by
+	// environment variable
+	Path string `dials:"CONFIGPATH"`
+	Val1 int    `dials:"Val1"`
+	Val2 string `dials:"Val2"`
+}
+
+// ConfigPath reflects where the path to config file is stored. Path field
+// will be populated from environment variable
+func (c *Config) ConfigPath() (string, bool) {
+	return c.Path, true
+}
+
+func TestYAMLConfigEnvFlag(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	envErr := os.Setenv("CONFIGPATH", "../testhelper/testconfig.yaml")
+	require.NoError(t, envErr)
+	defer os.Unsetenv("CONFIGPATH")
```

Comment:
Let's add a doc comment to this test saying that it can't run concurrently with any other test because of env manipulation.